src/client/components/AddReferenceLayerModal.tsx

```diff
@@ -0,0 +1,504 @@
+  readonly project: ProjectId | null;
+  readonly layer: ReferenceLayerGeojson | null;
+  readonly numberOfFeatures: number;
+  readonly label_field: string | null;
+  readonly layer_type: ReferenceLayerTypes.Point | ReferenceLayerTypes.Polygon | null;
+  readonly fields?: readonly string[] | null;
+  readonly valid: false;
+}
+
+interface ConfigurableForm {
+  readonly name: string | null;
+  readonly label_field: string | null;
+  readonly project: ProjectId;
+  readonly layer: ReferenceLayerGeojson | null;
+  readonly layer_type: ReferenceLayerTypes.Point | ReferenceLayerTypes.Polygon | null;
+  readonly fields?: readonly string[] | null;
+  readonly numberOfFeatures: number | null;
+}
+
+const AddReferenceLayerModal = ({
+  project,
+  showModal
+}: {
+  readonly project: IProject;
+  readonly showModal: boolean;
+}) => {
+  const hideModal = () => store.dispatch(toggleReferenceLayersModal()) && resetForm();
+  const [fileError, setFileError] = useState<string | undefined>();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const importNumberRef = useRef(0);
+
+  async function checkCsvContainsValidPoints(file: Blob): Promise<boolean> {
+    const contents = await new Response(file).text();
+    const lines = contents.split("\n");
+    if (lines.length > 0) {
+      const header = lines[0].replace("\r", "").split(",");
+      if (
+        (header.includes("lat") || header.includes("latitude")) &&
+        (header.includes("lon") || header.includes("long") || header.includes("longitude"))
+      ) {
+        return true;
+      } else {
+        setFileError("CSV must include latitude and longitude");
+        return false;
+      }
+    } else {
+      setFileError("CSV must include at least one record");
+      return false;
+    }
+  }
+
+  const blankForm: ConfigurableForm = {
+    name: null,
+    label_field: null,
+    project: project.id,
+    layer: null,
+    layer_type: null,
+    numberOfFeatures: null
+  };
+
+  const [importResource, setImportResource] = useState<
+    WriteResource<ProjectId, ReferenceLayerGeojson>
+  >({
+    data: project.id
+  });
+
+  const [importResponse, setImportResponse] = useState<ReferenceLayerGeojson | undefined>(
+    undefined
+  );
+
+  const [createLayerResource, setCreateLayerResource] = useState<
+    WriteResource<ConfigurableForm, ReferenceLayerWithGeojson>
+  >({
+    data: blankForm
+  });
+  const formData = createLayerResource.data;
+
+  function resetForm() {
+    setImportResource({
+      data: project.id
+    });
+    setImportResponse(undefined);
+    setFileError(undefined);
+    setCreateLayerResource({ data: blankForm });
+  }
+
+  function onReaderLoadCsv(event: ProgressEvent<FileReader>) {
+    const csvText = event.target?.result && event.target.result.toString();
+    const results = csvText && readString(csvText, { header: true });
+    // @ts-ignore
+    const geojson = results && convertCsvToGeojson(results);
+    geojson && setImportResponse(geojson);
```

Comment:
Let's transform the headers to be in lowercase. Based on my brief search for sample data to test this feature with, it seems like ALL_UPPERCASE is a real favorite of Open Data Portal ™️  data sources, so it would be nicer to be case-insensitive here.
```suggestion
    const csvText = event.target?.result && event.target.result.toString();
    const results = csvText && readString(csvText, { header: true, transformHeader: s => s.toLowerCase() }); ,
    const geojson = results && convertCsvToGeojson(results);
    geojson && setImportResponse(geojson);
```